### PR TITLE
Add wireguard config

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

Add wireguard config

### Changes

- Add wireguard checkbox in VM, Full VM
- Create getWireguard function

### Related Issues

- https://github.com/threefoldtech/grid_weblets/issues/1326

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [ ] Code format and docstrings

### CODE REVIEW

**Comment by zaelgohary** [comment]
Build flow used to fail because I imported grid client the wrong way.

File: src/elements/fullvm/Fullvm.wc.svelte [hunk 1]
```diff
@@ -48,6 +48,7 @@
     { label: "Public IPv4", symbol: "publicIp", placeholder: "", type: 'checkbox' },
     { label: "Public IPv6", symbol: "publicIp6", placeholder: "", type: 'checkbox' },
     { label: "Planetary Network", symbol: "planetary", placeholder: "", type: 'checkbox' },
+    { label: "Wireguard Config", symbol: "wireguard", placeholder: "", type: 'checkbox' },
```
Comment (AhmedHanafy725) [inline src/elements/fullvm/Fullvm.wc.svelte]:
```suggestion
    { label: "Add Wireguard Access", symbol: "wireguard", placeholder: "", type: 'checkbox' },
```

File: src/elements/kubernetes/Kubernetes.wc.svelte [hunk 2]
```diff
@@ -59,6 +59,7 @@
     { label: "Public IPv4", symbol: "publicIp", type: 'checkbox' },
     { label: "Public IPv6", symbol: "publicIp6", type: 'checkbox' },
     { label: "Planetary Network", symbol: "planetary", placeholder: "Enable planetary network", type: 'checkbox' },
+    { label: "Wireguard Config", symbol: "wireguard", placeholder: "", type: 'checkbox' },
```
Comment (AhmedHanafy725) [inline src/elements/kubernetes/Kubernetes.wc.svelte]: no need to add it to kubernetes, it should be enabled by default

File: src/elements/vm/Vm.wc.svelte [hunk 3]
```diff
@@ -49,6 +49,7 @@
     { label: "Public IPv4", symbol: "publicIp", placeholder: "", type: 'checkbox' },
     { label: "Public IPv6", symbol: "publicIp6", placeholder: "", type: 'checkbox' },
     { label: "Planetary Network", symbol: "planetary", placeholder: "", type: 'checkbox' },
+    { label: "Wireguard Config", symbol: "wireguard", placeholder: "", type: 'checkbox' },
```
Comment (AhmedHanafy725) [inline src/elements/vm/Vm.wc.svelte]:
```suggestion
    { label: "Add Wireguard Access", symbol: "wireguard", placeholder: "", type: 'checkbox' },
```

File: src/types/kubernetes.ts [hunk 4]
```diff
@@ -43,10 +43,15 @@ export class Master extends Base {
...
-  constructor(public name: string = "NW" + v4().split("-")[0], public ipRange: string = "10.20.0.0/16") {}
+  constructor(
+    public name: string = "NW" + v4().split("-")[0],
+    public ipRange: string = "10.20.0.0/16",
+    public addAccess: boolean = false,
+  ) {}
 
   public get valid(): boolean {
     const { name, ipRange } = this;
+    console.log("addAccess", this.addAccess);
```
Comment (AhmedHanafy725) [inline src/types/kubernetes.ts]: please remove the debugging logs

File: src/utils/createNetwork.ts [hunk 5]
```diff
@@ -1,9 +1,32 @@
...
   const network = new window.configs.grid3_client.NetworkModel();
   network.name = nw.name;
   network.ip_range = nw.ipRange;
-  network.addAccess = access;
+  network.addAccess = nw.addAccess;
+
+  if (network?.addAccess && network?.name) {
+    const grid3 = await getClient();
+
+    await grid3.networks.getWireGuardConfigs({ name: network.name });
```
Comment (AhmedHanafy725) [inline src/utils/createNetwork.ts]: why do we get the wireguard config while creating the network object?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: src/utils/createNetwork.ts (hunk 5 above)
Comment (zaelgohary) [reply src/utils/createNetwork.ts]: I separated getwireguardConfig into a new file. now, the user can check his configs in his deployed list.

File: src/utils/deployKubernetes.ts [hunk 6]
```diff
@@ -10,7 +10,7 @@ export default async function deployKubernetes(data: Kubernetes, profile: IProfi
 
   const masterNodes = [createNode(master)];
   const workerNodes = workers.map(createNode);
-  const _network = createNetwork(network, true);
+  const _network = createNetwork(network);
```
Comment (AhmedHanafy725) [inline src/utils/deployKubernetes.ts]: for kubernetes, it should be true be default

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**